### PR TITLE
PR-Firestore-Request-Details

### PR DESCRIPTION
**Pull request by AntonioRdzNav**
This is the **6th PR** created to review the code of my intern project.

This PR focuses on adding the **basic structure of the RequestDetails** view. The content of the view (Header, CodeViewer, and InspectionSection) is empty for now, and it will be added on a separate PR.

About to click a row on the requests table:

image

Redirected to the request details view after clicking the table row:

image

### CODE REVIEW

File: src/components/Firestore/Requests/RequestDetails/index.tsx [hunk 1]
```diff
@@ -14,19 +14,79 @@
...
+    requestMethod,
+    resourcePath,
+    outcomeData,
+  ] = useRequestMainData(selectedRequest);
+  const [
+    firestoreRules,
+    linesOutcome,
+    linesIssues,
+    inspectionElements,
+  ] = useRequestDetailedData(selectedRequest);
```
Comment (yuchenshi) [inline src/components/Firestore/Requests/RequestDetails/index.tsx:58]: I don't think we need a function that returns a tuple here. `firestoreRules = selectedRequest?.data?.rules;` is easier to read IMO. `inspectionElement` can probably use its own helper function (which can live in the same file).

**Review by yuchenshi — APPROVED** [review]

**Comment by yuchenshi** [comment]
And happy 2021! Please kindly use the year `2021` in copyright headers for any new files. (No need to touch existing files.)

**Review by samhorlbeck — APPROVED** [review]
Approval pending fixing the copyright headers

**Comment by AntonioRdzNav** [comment]
All copyright headers from **/Firestore/Requests/** files are **up-to-date** (2021), I will make sure the files from the next PRs are also up-to-date.

**Pull request merged**